### ISSUE TITLE
Should not list Carthage in Framework Search Paths

### ISSUE THREAD

**Issue by brentleyjones** [contentful/contentful.swift#101]
> - contentful.swift version number: 0.7.2
> - Xcode version number: 8.3.3
> - Target operating system(s) and version number(s)
    - [x] iOS: 
    - [ ] tvOS: 
    - [ ] watchOS:
    - [ ] macOS:
> - Package manager:
    - [x] Carthage
    - [ ] Cocoapods

Currently `FRAMEWORK_SEARCH_PATHS ` is set to `$(PROJECT_DIR)/Carthage/Build/iOS`. This gives a warning when building from source in a consuming workspace:

```
ld: warning: directory not found for option '-F/Users/bejones/Development/project/Carthage/Checkouts/contentful.swift/Carthage/Build/iOS'
```

The correct setup is to not have this and to include ObjectMapper in the Workspace and use Xcode's Implicit Dependency Management. That way it knows how to compile both as a part of Carthage and in the workspace for development: https://www.quora.com/How-should-an-OSS-library-be-structured-for-distribution-with-both-Carthage-and-CocoaPods/answer/Justin-Spahr-Summers

See https://github.com/RxSwiftCommunity/RxKeyboard/issues/25 and https://github.com/RxSwiftCommunity/RxKeyboard/pull/26 where I reported and fixed this for another project.

**Comment by loudmouth** [comment]
Hi @brentleyjones thanks for bringing this to my attention. It was not something I was aware of, and I'd love to confirm with you the correct solution as you seem knowledgeable.

So, rather than linking the framework search paths, I should create an xcworkspace file and include the Carthage checkouts in this workspace. If my understanding is correct, xcode will automatically build these sub projects and make them implicit dependencies for the main target. Is this correct?

**Comment by brentleyjones** [comment]
@loudmouth Correct. If the xcodeproj files are in the same workspace then Xcode will build them, if you link to one of their targets.

**Comment by loudmouth** [comment]
Thanks @brentleyjones 

I've made a fix in #102 and read the article you linked written by the Carthage maintainer.

The one thing I'm still not able to figure out is why `carthage build --no-skip-current` fails (and also why it builds test only dependencies listed in Cartfile.private).

Assuming I have done everything correctly in my PR, shouldn't the `carthage build` succeed?

**Comment by brentleyjones** [comment]
Yes. I'll pull down the PR and see what the issue is.

**Comment by loudmouth** [comment]
thanks :-)

**Comment by brentleyjones** [comment]
Running `carthage bootstrap --no-build` first then `carthage build --no-skip-current` worked. Needed to bootstrap in order to have the sources for the dependencies available. Personally with this solution I recommend having the dependencies setup as submodules so people can checkout the project, open the workspace, and build, without having to install or know anything about Carthage. (If you want to, an easy way to set them up is to delete your `Carthage/Checkout` folder and run `carthage bootstrap --use-submodules --no-build`)

**Comment by loudmouth** [comment]
Awesome. I will look into this first thing tomorrow morning.

I really appreciate that you looked into this and checked out the PR—I'm always striving for developer friendly project configuration.

**Comment by loudmouth** [comment]
Fixed in #102
